src/main/kotlin/app/revanced/patcher/fingerprint/method/impl/MethodFingerprint.kt

```diff
@@ -42,6 +44,93 @@ abstract class MethodFingerprint(
+                if (accessFlags and AccessFlags.CONSTRUCTOR.value != 0) {
+                    // Constructors always have void return type
+                    returnTypeValue = "V"
+                } else {
+                    return ""
+                }
+            }
+
+            return buildString {
+                append(accessFlags)
+                append(returnTypeValue.first())
+                parameters?.forEach { append(it.first()) }
+            }
+        }
+
+        internal fun createMethodLookupMap(classes: Iterable<ClassDef>) : Map<String, List<Pair<ClassDef, Method>>> {
+            fun addMethodToMapList(map: MutableMap<String, MutableList<Pair<ClassDef, Method>>>,
+                                   key:String, classDef : ClassDef, method: Method) {
+                var list = map[key]
+                if (list == null) {
+                    list = LinkedList()
+                    map[key] = list
+                }
+                list += Pair(classDef, method)
+            }
+
+            val methodMap = mutableMapOf<String, MutableList<Pair<ClassDef, Method>>>()
+            for (classDef in classes) {
+                for (method in classDef.methods) {
+                    val accessFlagsReturnKey = method.accessFlags.toString() + method.returnType.first()
+                    val accessFlagsReturnParametersKey = buildString {
+                        append(accessFlagsReturnKey)
+                        method.parameterTypes.forEach { append(it.first()) }
+                    }
+
+                    // For signatures with no access or return type specified
+                    addMethodToMapList(methodMap, "", classDef, method)
+                    // Only access and return type
+                    addMethodToMapList(methodMap, accessFlagsReturnKey, classDef, method)
+                    // Access, return, and parameter count
+                    addMethodToMapList(methodMap, accessFlagsReturnParametersKey, classDef, method)
+                    // Using more fine grained lookup is possible, but in practice it does not improve performance
+
+                    // Also allow looking up by Strings.
+                    // Escape the string with a prefix, to prevent any odd strings from matching a method signature key.
+                    method.implementation?.instructions?.forEach { instruction ->
+                        if (instruction.opcode == Opcode.CONST_STRING || instruction.opcode == Opcode.CONST_STRING_JUMBO) {
+                            val string = ((instruction as ReferenceInstruction).reference as StringReference).string
+                            addMethodToMapList(methodMap, "String:$string", classDef, method)
+                        }
+                    }
+                }
+            }
+            return methodMap
+        }
+
+        fun Iterable<MethodFingerprint>.resolve(context: BytecodeContext, methodMap: Map<String, List<Pair<ClassDef, Method>>>) {
```

Comment:
This extension function particularly extends `Iterable<MethodFingerprint>` only; this is arbitrary, instead, functions for resolving a single MethodFingerprint should exist.  Ideally, these extension functions are moved as functions in `MethodFingerprint` instead of extension functions.